### PR TITLE
adding a testing section and Python's Replayer

### PR DESCRIPTION
**Pull request by rachfop**
# Need Python's Replayer code sample checked

- [x] Chad to review: docs/python/how-to-replay-a-workflow-execution-in-python.md

## What does this PR do?

- Removed `Replay` from `observability` section to a new section called 'testing`.
- The `testing` section will be updated based on the [outcome of this issue.](https://github.com/temporalio/sdk-features/issues/113) - but that isn't a blocker for this PR.
- [x] Awaits this [Python PR](https://github.com/temporalio/sdk-python/pull/108) to merge before this PR can be merged.
- [x] Will need to verify the link of `replay_workflow()` once the Python API docs update.
- [x] Will need to verify the `To retrieve the Workflow History`section for **all** SDKs. That section isn't meant to be an exhaustive list, rather provides a few examples.

## Notes to reviewers
What is the name of this feature - I tried to use `Replay` and not `Replays` when referencing it as a noun.
When referencing the action/verb, it is lower case: `replay`.

### CODE REVIEW

File: docs/python/how-to-replay-a-workflow-execution-in-python.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+In the following example, `history_json_str` references the Workflow History as a JSON string.
+
+```python
+async def run_replayer(history_json_str: str):
+    replayer = Replayer(workflows=[YourWorkflow])
+    await replayer.replay_workflow(history_json_str)
+```
+
+`run_replayer()` will return an error if the Workflow History is non-deterministic.
```
Comment (djmagee) [inline docs/python/how-to-replay-a-workflow-execution-in-python.md:22]:
```suggestion
If the Workflow History is non-deterministic, `run_replayer()` returns an error.
```

File: docs/python/how-to-replay-a-workflow-execution-in-python.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+title: How to replay a Workflow Execution in Python
+sidebar_label: Replay a Workflow Execution
+description: Replay a Workflow Execution
+tags:
+  - developer-guide
+  - sdk
+  - python
+---
+
+To Replay a Workflow Execution use the [`replay_workflow()`](https://python.temporal.io/temporalio.worker.replayer#replay_workflow) method and pass a Workflow History as an argument.
```
Comment (djmagee) [inline docs/python/how-to-replay-a-workflow-execution-in-python.md:12]:
```suggestion
To replay a Workflow Execution, use the [`replay_workflow()`](https://python.temporal.io/temporalio.worker.replayer#replay_workflow) method and pass a Workflow History as an argument.
```

File: docs/python/how-to-replay-a-workflow-execution-in-python.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+async def run_replayer(history_json_str: str):
+    replayer = Replayer(workflows=[YourWorkflow])
+    await replayer.replay_workflow(history_json_str)
+```
+
+`run_replayer()` will return an error if the Workflow History is non-deterministic.
+
+:::note
+
+If the Workflow History is exported by the [Temporal Web UI](/web-ui) or through the [tctl](/tctl), you can pass the JSON file history object as a JSON string or as a python dictionary, through the `json.load()` function, which takes a file object and returns the JSON object.
```
Comment (djmagee) [inline docs/python/how-to-replay-a-workflow-execution-in-python.md:26]:
```suggestion
If the Workflow History is exported by [Temporal Web UI](/web-ui) or through [tctl](/tctl), you can pass the JSON file history object as a JSON string or as a Python dictionary through the `json.load()` function, which takes a file object and returns the JSON object.
```

File: guide-gen/guide-configs/app-dev/testing.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
+{
+  "file_name": "testing.md",
+  "id": "testing",
+  "file_dir": "application-development",
+  "title": "Application development - Testing",
+  "sidebar_label": "Testing",
+  "description": "The Testing section of the Temporal Application development guide covers the many ways to test the state of your Temporal Application – that is, ways to view what Workflow Executions are tracked by the Platform and the state of any given Workflow Execution, either currently or at points of an execution.",
```
Comment (djmagee) [inline guide-gen/guide-configs/app-dev/testing.json:7]:
```suggestion
  "description": "The Testing section of the Temporal Application development guide covers the many ways to test the state of your Temporal Application; that is, ways to view which Workflow Executions are tracked by the Platform and the state of any given Workflow Execution, either currently or at points of an execution.",
```

File: docs/python/how-to-replay-a-workflow-execution-in-python.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+---
+id: how-to-replay-a-workflow-execution-in-python
+title: How to replay a Workflow Execution in Python
+sidebar_label: Replay a Workflow Execution
+description: Replay a Workflow Execution
```
Comment (djmagee) [inline docs/python/how-to-replay-a-workflow-execution-in-python.md:5]:
```suggestion
description: To replay a Workflow Execution, use the `replay_workflow()` method and pass a Workflow History as an argument.
```

File: docs/app-dev-context/replays.md [hunk 6]
```diff
@@ -1,12 +1,24 @@
...
   - guide-context
 ---
 
-Replays recreate the exact state the Workflow code was in. You can replay Workflows from the beginning of their history when resumed.
+Replay recreates the exact state of the Workflow code. You can replay Workflows from the beginning of their history when resumed.
 
 Replays allow code to resume only if it is compatible from a deterministic point of view.
+
+**To retrieve the Workflow History**
+You can retrieve the Workflow History from any of the following options and then pass the object to your SDK of choice.
```
Comment (djmagee) [inline docs/app-dev-context/replays.md:15]:
```suggestion
To retrieve the Workflow History, use any of the following options and then pass the object to your SDK of choice.
```
Alternatively, add a blank line after the bold psuedoheading.

File: docs/app-dev-context/replays.md [hunk 7]
```diff
@@ -1,12 +1,24 @@
 ---
-id: replays
+id: replay
 title: How to Replay a Workflow Execution
-description: Replays recreate the exact state the Workflow code was in.
-sidebar_label: Replays
+description: Replay recreates the exact state of the Workflow code.
```
Comment (djmagee) [inline docs/app-dev-context/replays.md:4]:
```suggestion
description: Replay recreates the exact state of a Workflow Execution.
```
I have a mini-Dominik sitting on my shoulder.

File: docs/app-dev-context/replays.md [hunk 8]
```diff
@@ -1,12 +1,24 @@
...
+sidebar_label: Replay
 tags:
   - guide-context
 ---
 
-Replays recreate the exact state the Workflow code was in. You can replay Workflows from the beginning of their history when resumed.
+Replay recreates the exact state of the Workflow code. You can replay Workflows from the beginning of their history when resumed.
 
 Replays allow code to resume only if it is compatible from a deterministic point of view.
```
Comment (djmagee) [inline docs/app-dev-context/replays.md:12]:
```suggestion
Replay recreates the exact state of a Workflow Execution.
You can replay a Workflow from the beginning of its history when resumed.

Replays allows code to resume only if it is compatible from a deterministic point of view.
```

File: docs/app-dev-context/replays.md [hunk 9]
```diff
@@ -1,12 +1,24 @@
...
 tags:
   - guide-context
 ---
 
-Replays recreate the exact state the Workflow code was in. You can replay Workflows from the beginning of their history when resumed.
+Replay recreates the exact state of a Workflow Execution.
+You can replay a Workflow from the beginning of its history when resumed.
 
-Replays allow code to resume only if it is compatible from a deterministic point of view.
+Replays allows code to resume only if it is compatible from a deterministic point of view.
```
Comment (djmagee) [inline docs/app-dev-context/replays.md:13]:
```suggestion
Replay allows code to resume only if it is compatible from a deterministic point of view.
```
Oops. My bad.

File: docs/app-dev-context/replays.md [hunk 10]
```diff
@@ -1,12 +1,24 @@
...
+
+To retrieve the Workflow History, use any of the following options and then pass the object to your SDK of choice.
+
+- Proto History object
+- [Temporal Web UI](/web-ui)
+  1. Select a Workflow ID.
+  2. Click **Download**.
+- [tctl](/tctl) commands, for example:
```
Comment (djmagee) [inline docs/app-dev-context/replays.md:21]:
```suggestion
- [tctl](/tctl) commands; for example:
```

File: docs/python/how-to-replay-a-workflow-execution-in-python.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+In the following example, `history_json_str` references the Workflow History as a JSON string.
+
+```python
+async def run_replayer(history_json_str: str):
+    replayer = Replayer(workflows=[YourWorkflow])
+    await replayer.replay_workflow(history_json_str)
+```
+
+If the Workflow History is non-deterministic, `run_replayer()` returns an error.
```
Comment (cretz) [inline docs/python/how-to-replay-a-workflow-execution-in-python.md:22]:
```suggestion
If the Workflow History is non-deterministic, `run_replayer()` raises an error.
```

**Review by cretz — APPROVED** [review]

**Review by djmagee — APPROVED** [review]

**Pull request merged**